packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -0,0 +1,285 @@
+import assert from 'assert';
+import { MappingTemplate } from '@aws-amplify/graphql-transformer-core';
+import { TransformerContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { DynamoDbDataSource } from '@aws-cdk/aws-appsync';
+import { Table } from '@aws-cdk/aws-dynamodb';
+import * as cdk from '@aws-cdk/core';
+import { ObjectTypeDefinitionNode } from 'graphql';
+import {
+  bool,
+  compoundExpression,
+  DynamoDBMappingTemplate,
+  equals,
+  Expression,
+  ifElse,
+  iff,
+  list,
+  nul,
+  obj,
+  ObjectNode,
+  or,
+  print,
+  raw,
+  ref,
+  set,
+  str,
+} from 'graphql-mapping-template';
+import {
+  applyCompositeKeyConditionExpression,
+  applyKeyConditionExpression,
+  attributeTypeFromScalar,
+  ModelResourceIDs,
+  NONE_VALUE,
+  ResourceConstants,
+  toCamelCase,
+} from 'graphql-transformer-common';
+import { HasManyDirectiveConfiguration, HasOneDirectiveConfiguration } from './types';
+import { getConnectionAttributeName } from './utils';
+
+export function makeGetItemConnectionWithKeyResolver(config: HasOneDirectiveConfiguration, ctx: TransformerContextProvider) {
+  const { connectionFields, field, fields, object, relatedType, relatedTypeIndex } = config;
+  assert(relatedTypeIndex.length > 0);
+  const localFields = fields.length > 0 ? fields : connectionFields;
+  const table = getTable(ctx, relatedType);
+  const { keySchema } = table as any;
+  const dataSource = ctx.api.host.getDataSource(`${relatedType.name.value}Table`);
+  const partitionKeyName = keySchema[0].attributeName;
+  const keyObj = {
+    [partitionKeyName]: ref(`util.dynamodb.toDynamoDBJson($util.defaultIfNullOrBlank($ctx.source.${localFields[0]}, "${NONE_VALUE}"))`),
+  };
+
+  // Add a composite sort key or simple sort key if there is one.
+  if (relatedTypeIndex.length > 2) {
+    const rangeKeyFields = localFields.slice(1);
+    const sortKeyName = keySchema[1].attributeName;
+    const condensedSortKeyValue = condenseRangeKey(rangeKeyFields.map(keyField => `\${ctx.source.${keyField}}`));
+
+    keyObj[sortKeyName] = ref(`util.dynamodb.toDynamoDBJson($util.defaultIfNullOrBlank("${condensedSortKeyValue}", "${NONE_VALUE}"))`);
+  } else if (relatedTypeIndex.length === 2) {
+    const sortKeyName = keySchema[1].attributeName;
+
+    keyObj[sortKeyName] = ref(`util.dynamodb.toDynamoDBJson($util.defaultIfNullOrBlank($ctx.source.${localFields[1]}, "${NONE_VALUE}"))`);
+  }
+
+  const resolver = ctx.resolvers.generateQueryResolver(
+    object.name.value,
+    field.name.value,
+    dataSource as any,
+    MappingTemplate.s3MappingTemplateFromString(
+      print(
+        ifElse(
+          or(localFields.map(f => raw(`$util.isNull($ctx.source.${f})`))),
+          raw('#return'),
+          compoundExpression([
+            DynamoDBMappingTemplate.getItem({
```

Comment:
Can we also make this a query similar to how it's for auth vNext since we'll apply an auth filter here as well to protect the connection.